python/whylogs/core/constraints/metric_constraints.py

```diff
@@ -171,19 +185,109 @@ def validate_profile(self, dataset_profile: DatasetProfileView) -> Tuple[bool, O
         return (validate_result, metric_summary)
 
 
+@dataclass
```

Comment:
Are `MetricConstraint` and `DatasetConstraint` getting value from being `@dataclass` ?

Reply:
The ctors are nice. The `Callable`s will be the hard part of [de]serialization though, so not super-helpful there.

Reply:
Probably should be frozen